### ISSUE TITLE
Issue with NO WAIT mode

### ISSUE THREAD

**Issue by JuniorBaumgart** [hgourvest/node-firebird#355]
Hi, i am writting a code with firebird 2.5 and node-firebird 1.1.9
On my database.js im doing the conection this way:

const firebird = require("node-firebird");

const options = {
  host: "localhost",
  port: 3050,
  database: "TEST",
  user: "SYSDBA",
  password: "masterkey",
  lowercase_keys: true,
};

module.exports = {
  query: (sql, params = []) =>
    new Promise((resolve, reject) => {
      console.log("Executando SQL:", sql);
      console.log("Com parâmetros:", params);

      firebird.attach(options, (err, db) => {
        if (err) return reject(new Error(`Erro ao conectar ao banco: ${err.message}`));

        const transactionOptions = {
          isolation: firebird.ISOLATION_READ_COMMITTED,
          lockTimeout: 10,
        };

        db.startTransaction(transactionOptions, (err, transaction) => {
          if (err) {
            db.detach();
            return reject(new Error(`Erro ao iniciar transação: ${err.message}`));
          }

         
          transaction.query(sql, params, (err, result) => {
            if (err) {
              console.error("Erro ao executar a query:", err);
              transaction.rollback(() => {
                db.detach();
                reject(new Error(`Erro ao executar query: ${err.message}`));
              });
              return;
            }

            console.log("Resultado da consulta:", result);

           
            transaction.commit((commitErr) => {
              if (commitErr) {
                console.error("Erro ao fazer commit:", commitErr);
                transaction.rollback(() => {
                  db.detach();
                  reject(new Error(`Erro ao fazer commit: ${commitErr.message}`));
                });
                return;
              }

              db.detach();
              resolve(result);
            });
          });
        });
      });
    }),
};

However, when checking with the isql tool, I identified that the mon$lock_timeout field is -1 (WAIT)
i really need help with that

**Comment by maratth** [comment]
Hi,

First your `transactionOptions` isn't good. `db.startTransaction` take only isolation array in first parameter.
However, currently this driver do not expose a isolation level for nowait transaction.

I think you can do by using custom isolation.
Ex : 
```js
const FirebirdConst = require('node-firebird/lib/wire/const');

const isolation = [
	FirebirdConst.isc_tpb_version3,
	FirebirdConst.isc_tpb_write,
	FirebirdConst.isc_tpb_nowait,
	FirebirdConst.isc_tpb_read_committed,
	FirebirdConst.isc_tpb_no_rec_version,
];

Firebird.attach(option, (err, db) => {
  db.transaction(isolation, (err, transaction) => {
    // Do anything
  });
});
```

With this custom isolation, you can start nowait transaction.
